libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Properties.hs

```diff
@@ -479,26 +533,34 @@ genDCerts = do
         case dc of
           DCertDeleg d
             | RegKey regCred <- d ->
-                if regCred `Set.member` regCreds
+                if regCred `hasKey` regCreds
                   then pure (dcs, ss, regCreds)
-                  else pure (dc : dcs, ss, Set.insert regCred regCreds)
-            | DeRegKey deregCred <- d ->
-                if deregCred `Set.member` regCreds
-                  then
-                    insertIfNotPresent dcs (Set.delete deregCred regCreds) Nothing
+                  else pure (dc : dcs, ss, Map.insert regCred (Coin 99) regCreds) -- 99 is a NonZero Value
```

Comment:
Would it make sense to generate a `NonZero` values instead of hard coding a `99`?

Reply:
We do not use this map for anything other than testing if some key is in the map, and in one place if the value of the key is non-zero. It seems like overkill to generate a random non-zero value, because this map is going to be garbage collected.  It is clear we are inserting a non zero value and that is all we need.